RandBLAS/trig_skops.hh

```diff
@@ -316,125 +298,157 @@ namespace RandBLAS {
 
     template <typename T>
     void fht_dispatch(
-        bool left,
+        bool left, // Pre-multiplying?
         blas::Layout layout,
-        T* buff,
-        int64_t log_n,
         int64_t num_rows,
-        int64_t num_cols
+        int64_t num_cols,
+        int64_t log_n,
+        T* A
         )
     {
         if(left && layout == blas::Layout::ColMajor)
-            fht_left_col_major(buff, log_n, num_rows, num_cols);
+            fht_left_col_major(A, log_n, num_rows, num_cols);
         else if(left && layout == blas::Layout::RowMajor)
-            fht_left_row_major(buff, log_n, num_rows, num_cols);
+            fht_left_row_major(A, log_n, num_rows, num_cols);
         else if(!left && layout == blas::Layout::ColMajor)
-            fht_right_col_major(buff, log_n, num_rows, num_cols);
+            fht_right_col_major(A, log_n, num_rows, num_cols);
         else
-            fht_right_row_major(buff, log_n, num_rows, num_cols);
+            fht_right_row_major(A, log_n, num_rows, num_cols);
     }
 }
 
 
 namespace RandBLAS::trig {
+template <SignedInteger sint_t = int64_t>
+struct HadamardMixingOp{
+    sint_t* diag_scale;
+    sint_t* selected_idxs;
+    blas::Layout layout;
+    int64_t m;
+    int64_t n;
+    int64_t d;
+    bool left;
+    bool filled = false; // will be updated by `miget`
+
+    // Constructor
+    HadamardMixingOp(bool left,
+                     blas::Layout layout,
+                     int64_t m,
+                     int64_t n,
+                     int64_t d
+                    ) : left(left), layout(layout), m(m), n(n), d(d) {
+        if(left)
+            diag_scale = new sint_t[m];
+        else
+            diag_scale = new sint_t[n];
+        selected_idxs = new sint_t[d];
+    }
+
+    // Destructor
+    ~HadamardMixingOp() {
+        free(this->diag_scale);
+        free(this->selected_idxs);
+    }
+
+    private:
+};
+
+/*
+ * A free-function that performs an inversion of a matrix transformed by `lmiget | rmiget`
+ * the inversion is also performed in-place
+*/
+template <typename T, SignedInteger sint_t = int64_t>
+void invert(
+    HadamardMixingOp<sint_t> &hmo, // details about the transform
+    T* SA // sketched matrix
+) {
```

Comment:
This type of function name would be problematic if someone wanted to call it from a language that doesn't support function overloading. Please propose some alternative names (here on a comment thread is fine).

Reply:
As of right now, I renamed it to `invert_hadamard` --- but maybe since its a free method of the `HadamardMixingOp` class, we can also call it `invert_hmo`?